### PR TITLE
improve partial Codecov

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
Partial work on #463 increasing to 91%

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by williamFalcon** [comment]
@Borda no tensorboardx... keep it pytorch summaryWriter please

**Comment by Borda** [comment]
@bmartinn could you help why the Trains tests are failing?

**Comment by bmartinn** [comment]
@Borda sure, I'll take a look. 
I think I have an idea on what the problem is, I'll work on a fix.

**Comment by bmartinn** [comment]
@Borda I just pushed a fix as a PR on the Codecov branch

**Comment by Borda** [comment]
@bmartinn merger you PR here but the failer is the same?
is seems that it complains about existing names or so?
Also curious, does it work also offline?

**Comment by bmartinn** [comment]
Hi @Borda,

It seems like two Loggers are running simultaneously, is this possible ?

*Trains* is all about logging all the outputs/logs/console/artifacts to the *trains-server* (usually self hosted). So yes, currently there is no offline mode. That said I could easily add "bypass" argument, making sure the interface works, and it will just bypass sending data to the *trains-server*. 

What do you think?

**Comment by Borda** [comment]
@bmartinn Well we have not changed anything on CI side and it was passing before..
In particular, we are sure that Drone runs only one ob in concurrency, @luiscape ring?

**Comment by bmartinn** [comment]
Hmmm, anyhow it might be safer to have "bypass_mode"...
@Borda, I can PR a commit on the Codecov branch if you like

**Comment by Borda** [comment]
@bmartinn that would be great! Thx

**Comment by bmartinn** [comment]
Done :)

File: pytorch_lightning/trainer/distrib_data_parallel.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ def set_distributed_mode(self, distributed_backend, num_gpu_nodes):
                 self.use_ddp2 = False
 
         # throw error to force user ddp or ddp2 choice
-        if num_gpu_nodes > 1 and not (self.use_ddp2 or self.use_ddp):  # pragma: no cover
+        if num_gpu_nodes > 1 and not (self.use_ddp2 or self.use_ddp):  # pragma: no-cover
```
Comment (neggert) [inline pytorch_lightning/trainer/distrib_data_parallel.py]: Why `no-cover` here?

File: pytorch_lightning/trainer/distrib_parts.py [hunk 2]
```diff
@@ -511,7 +511,7 @@ def dp_train(self, model):
         # check for this bug (amp + dp + !01 doesn't work)
         # https://github.com/NVIDIA/apex/issues/227
         if self.use_dp and self.use_amp:
-            if self.amp_level == 'O2':  # pragma: no cover
+            if self.amp_level == 'O2':  # pragma: no-cover
```
Comment (neggert) [inline pytorch_lightning/trainer/distrib_parts.py]: Why `no-cover` here?

File: pytorch_lightning/trainer/evaluation_loop.py [hunk 3]
```diff
@@ -248,7 +248,7 @@ def evaluate(self, model, dataloaders, max_batches, test_mode: bool = False):
                 dataloader = dataloader.per_device_loader(device)
 
             for batch_idx, batch in enumerate(dataloader):
-                if batch is None:  # pragma: no cover
+                if batch is None:  # pragma: no-cover
```
Comment (neggert) [inline pytorch_lightning/trainer/evaluation_loop.py]: Why `no-cover` here?

**Comment by Borda** [comment]
> Looks good, but at some point we should do an audit of lines that have `pragma: no-cover`. IMO some of theses should be included in coverage.

I do agree, we need to work more on tests, but with this kind of commented lines and functional visual codecov UI it is much easier to spot weakness in the package...
Note that this is just one of several upcoming PRs on getting better coverage but I would like to keep it in smaller chunks :]

File: pytorch_lightning/trainer/distrib_data_parallel.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/trainer/distrib_data_parallel.py]: it was already there, but I ll drop it

File: pytorch_lightning/trainer/distrib_parts.py (hunk 2 above)
Comment (Borda) [reply pytorch_lightning/trainer/distrib_parts.py]: it was already there, but I ll drop it

File: pytorch_lightning/trainer/evaluation_loop.py (hunk 3 above)
Comment (Borda) [reply pytorch_lightning/trainer/evaluation_loop.py]: it was already there, but I ll drop it

**Pull request merged**